api/src/main/java/org/openmrs/module/fhir/api/impl/PractitionerServiceImpl.java

```diff
@@ -148,4 +159,46 @@ public Practitioner getPractitioner(String id) {
 	private List<Provider> searchProvidersByQuery(String query) {
 		return Context.getProviderService().getProviders(query, null, null, null, false);
 	}
+	
+	/**
+	 * @see org.openmrs.module.fhir.api.PractitionerService#createFHIRPractitioner(String)
+	 */
+	public Practitioner createFHIRPractitioner(Practitioner practitioner) {
+		Provider provider=null;
+		// If the representation has unknown gender and the birthday is null, that means the practitioner is not associated with any person resource. Because those attributes are compulsary if there is a person.
+		if (practitioner.getGender().equalsIgnoreCase(String.valueOf(AdministrativeGenderEnum.UNKNOWN))&&practitioner.getBirthDate()==null){ 
+			provider=new Provider();			
+			HumanNameDt humanNameDt= practitioner.getName(); // practitioners have only one name
+			if (humanNameDt.getUse() != null) {
+				String getUse = humanNameDt.getUse();
+				if (String.valueOf(NameUseEnum.USUAL).equalsIgnoreCase(getUse)) { // this values always set to usual
+					List<StringDt> givenNames = humanNameDt.getGiven();
+					StringDt givenName = givenNames.get(0);
+					provider.setName(valueOf(givenName));
+				}
+			}
+			List<IdentifierDt> identifiers=practitioner.getIdentifier();
+			IdentifierDt idnt=identifiers.get(0);
+			provider.setIdentifier(idnt.getValue());			
+			Provider omrsProvider = Context.getProviderService().saveProvider(provider); // create a practitioner
+			return FHIRPractitionerUtil.generatePractitioner(omrsProvider); 
+		}else{  // create a practitoner associated with a person resource. Either a new person or existing person.
+			Person person=FHIRPractitionerUtil.generateOpenMRSPerson(practitioner);
+			provider = new Provider();		
+			
+			/*if (practitioner.getId() != null) {
+				provider.setUuid(practitioner.getId().getIdPart());
+			}*/ 
+			// Don't set this since this is POST 
+			
+			List<IdentifierDt> identifiers=practitioner.getIdentifier();
+			if (identifiers != null) {
+				IdentifierDt idnt=identifiers.get(0);
+				provider.setIdentifier(idnt.getValue());
+			} 		 // Is it good to set the Identifier in a POST operation? But this is mandotory in OpenMRS. 
```

Comment:
No I meant identifier not the uuid. But i think this would be ok, coz the resource URI is associated with the uuid , not with the identifier. So I think we are not violating any REST principles here :)